pallets/rmrk-core/src/functions.rs

```diff
@@ -184,7 +190,47 @@ impl<T: Config> Nft<T::AccountId, StringLimitOf<T>> for Pallet<T> {
 	}
 }
 
-impl<T: Config> Pallet<T> {
+impl<T: Config> Pallet<T>
+where
+	T: pallet_uniques::Config<ClassId = CollectionId, InstanceId = NftId>,
+{
+
+	pub fn nft_to_account_id<AccountId: Codec>(collection_id: CollectionId, nft_id: NftId) -> AccountId {
+		(b"RmrkNft/", collection_id, nft_id)
+			.using_encoded(|b| AccountId::decode(&mut TrailingZeroInput::new(b)))
+			.expect("Decoding with trailing zero never fails; qed.")
+	}
+
+	pub fn decode_nft_account_id<AccountId: Codec>(account_id: T::AccountId) -> Option<(CollectionId, NftId)> {
+		let (prefix, tuple, suffix) = account_id
+			.using_encoded(|mut b| {
+				let slice = &mut b;
+				let r = <([u8; 8], (CollectionId, NftId))>::decode(slice);
+				r.map(|(prefix, tuple)| (prefix, tuple, slice.to_vec()))
+			})
+			.ok()?;
+		// Check prefix and suffix to avoid collision attack
+		if &prefix == b"RmrkNft/" && suffix.iter().all(|&x| x == 0) {
+			Some(tuple)
+		} else {
+			None
+		}
+	}
+
+	pub fn lookup_root_owner(collection_id: CollectionId, nft_id: NftId) -> Result<(T::AccountId, (CollectionId, NftId)), Error<T>> {
+		let parent =
+			pallet_uniques::Pallet::<T>::owner(collection_id, nft_id);
+		// Check if parent returns None which indicates the NFT is not available
```

Comment:
What are the cases a NFT can have no parent?

Reply:
I had thought of a couple cases, but the second case would be if we did the "rootless" children and tried to perform a `lookup_root_owner`
1) NFT is already burned (Added a test for this one)
2) A case if an ancestor NFT were burned leaving the